packages/devtools/src/elementstore.ts

```diff
@@ -1,12 +1,23 @@
 import type { ElementHandle } from 'puppeteer-core/lib/cjs/puppeteer/common/JSHandle'
+import type { Frame } from 'puppeteer-core/lib/cjs/puppeteer/common/FrameManager'
 
 export default class ElementStore {
     private _index = 0
     private _elementMap: Map<string, ElementHandle> = new Map()
+    private _frameMap: Map<Frame, Set<string>> = new Map()
 
     set (elementHandle: ElementHandle) {
         const index = `ELEMENT-${++this._index}`
         this._elementMap.set(index, elementHandle)
+        const frame = elementHandle.executionContext().frame()
+        if (frame) {
```

Comment:
We usually always expect a frame right? What would be a scenario where frame is null here? I wonder if it makes sense to better throw in this case rather than silently ignoring this.

Reply:
Yeah, a frame should be there in all of the cases. We are extracting the `ExecutionContext` of an element, which is always an `ExecutionContext` of the element owner frame. `ExecutionContex` of the frame will be always associated with this frame [pptr doc](https://pptr.dev/#?product=Puppeteer&version=v8.0.0&show=api-executioncontextframe). So I'm not sure which kind of error I could possibly throw here, because even if an element will not have a frame for some reason it just will be preserved in store until the whole page will be reloaded/navigated.